src/utils/logging.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * Logging is a helper class to manage log files; they can be viewed in the
+ * Troubleshooting tab in the UI.
+ *
+ * Usage:
+ *
+ * import Logging from '.../logging';
+ *
+ * const topicLog = Logging('topic');
+ * await Logging.topic('Log string goes here');
+ * // Equivalent to `await topicLog.log('Log string goes here');
+ * // The class has a `path` member:
+ * fs.readFile(Logging.topic.path, ...);
+ */
+
+import fs from 'fs';
+import path from 'path';
+import stream from 'stream';
+
+import XDGAppPaths from 'xdg-app-paths';
+const paths = XDGAppPaths({ name: 'rancher-desktop' });
+const logDir = path.join(paths.runtime() || paths.data(), 'logs');
+
+fs.mkdir(logDir, { recursive: true }, () => undefined);
+
+interface Log {
+  (message: string): Promise<void>;
+  path: string;
+  stream: stream.Writable;
+}
+
+interface Module {
+  (topic: string): Log;
+  [topic: string]: Log;
+}
+
+const logging = function(topic: string) {
+  if (!(topic in logging)) {
```

Comment:
You're storing properties on the function itself, right? The only other time I've seen this is on the regexp object, which is also a function.